module/platform/OpenCR/HardwareIO/src/hardwareio/send_request.cpp

```diff
@@ -19,8 +19,8 @@ namespace module::platform::OpenCR {
 
         if (servos_dirty) {
             // Write data is split into two components
-            std::array<dynamixel::v2::SyncWriteData<DynamixelServoWriteDataPart1>, 20> data1;
-            std::array<dynamixel::v2::SyncWriteData<DynamixelServoWriteDataPart2>, 20> data2;
+            dynamixel::v2::SyncWriteData<DynamixelServoWriteDataPart1> data1[20];
+            dynamixel::v2::SyncWriteData<DynamixelServoWriteDataPart2> data2[20];
```

Comment:
You should be able to just do this
```suggestion
            dynamixel::v2::SyncWriteCommand<DynamixelServoWriteDataPart1, 20> command1;
            dynamixel::v2::SyncWriteCommand<DynamixelServoWriteDataPart1, 20> command2;
```
Then `command1.data[i]` is the `i`th `SyncWriteData` so you can just write the servo data directly into the `SyncWriteCommand` struct. If this doesn't work then we should look at fixing this since copying that data should be unnecessary.

Reply:
The fields in all the messages are const - is this important for messages for any particular reason or is it fine to not have it const to be able to do this?

Reply:
The array is pass by reference so there shouldn't be any copying right?

Reply:
It is passed by `const&`, so it will be copied. All the fields are `const` to prevent the message fields from being accidentally corrupted, but I think it should be ok for `data` to be non-const

Reply:
I changed it but now it doesn't work :shrug:

Reply:
I've added this problem to the issue. Unfortunately I won't be able to work on this again until I get back in two weeks, but we need to get it merged asap since it is critical to the robots working.